### PR TITLE
[Subtitles][WebVTT] Fixed "seconds" variable type to double

### LINKED ISSUE #21807: kodi20 webvtt Subtitles out of sync when using Netflex addon after recent update
### Describe the bug
Here is a clear and concise description of what the problem is:

When using Netflix addon, webvtt subtitles appear about 0.1-0.4 seconds faster.
Rolling back to previous kodi20 release before around Aug 20, no sync mismatch. 
Tested on Windows10 PC, Shield TV pro, amazon firestick 4k max.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Perfect sync.

## Actual Behavior

As above.

## Possible Fix

I don't know.

### To Reproduce

1. Install latest kodi20 release and Netflix addon.
2. check subtitles sync several times.(At first, it may seem that the subtitles are in sync. So plz test repeatedly and carefully 'cause very slight sync mismatch)

### Debuglog

The debuglog can be found [here:]
[kodi.log](https://github.com/xbmc/xbmc/files/9486314/kodi.log)

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [x] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] OSX
 - [x] Windows
 - [ ] Windows UWP

 - Operating system version/name: Windows 10 pro, Android 11, Android 9
 - Kodi version: kodi20 latest release

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

### PR DESCRIPTION
**Pull request by CastagnaIT**
## Description

Fixed "seconds" variable type oversight changed by #21763

## Motivation and context

attempt to fix #21807

## How has this been tested?

I will wait user feedback after make a test build because i am not able reproduce the problem in easy way
EDIT: confirmed works https://github.com/xbmc/xbmc/issues/21807#issuecomment-1236812560

## What is the effect on users?

Right subtitle timing

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/cores/VideoPlayer/DVDSubtitles/webvtt/WebVTTHandler.cpp [hunk 1]
```diff
@@ -1145,6 +1145,6 @@ double CWebVTTHandler::GetTimeFromRegexTS(CRegExp& regex, int indexStart /* = 1
   if (!regex.GetMatch(indexStart).empty())
     sHours = std::stoi(regex.GetMatch(indexStart).c_str());
   int sMinutes = std::stoi(regex.GetMatch(indexStart + 1).c_str());
-  int sSeconds = std::atof(regex.GetMatch(indexStart + 2).c_str());
+  double sSeconds = std::atof(regex.GetMatch(indexStart + 2).c_str());
```
Comment (enen92) [inline xbmc/cores/VideoPlayer/DVDSubtitles/webvtt/WebVTTHandler.cpp:1148]: maybe `std::stof` to avoid the conversion to c string?

Comment (CastagnaIT) [reply xbmc/cores/VideoPlayer/DVDSubtitles/webvtt/WebVTTHandler.cpp:1148]: oh yes yourright

**Review by enen92 — APPROVED** [review]

**Comment by enen92** [comment]
jenkins build and merge

**Comment by enen92** [comment]
@CastagnaIT please check jenkins failures:

```
14:23:14 /home/jenkins/workspace/Android-X86/xbmc/cores/VideoPlayer/DVDSubtitles/webvtt/WebVTTHandler.cpp:1148:21: error: implicit conversion increases floating-point precision: 'float' to 'double' [-Werror,-Wdouble-promotion]
14:23:17   double sSeconds = std::stof(regex.GetMatch(indexStart + 2));
14:23:17          ~~~~~~~~   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
14:23:17 1 error generated.
```

I guess you need to use `std::stod`

**Pull request merged**